verified_boot_app.md

```diff
@@ -11,8 +38,109 @@ Src: https://github.com/Kicksecure/kicksecure-wiki-backup/blob/6900fdfbdea8693fd
+     or to disable UEFI Secure Boot for that boot, but it must always offer at
+     least one successful path.
+
+<!--
+* Support for binaries signed by multiple keys: Yes. (UEFI already supports that.) (This is important to allow the distribution or customer to cycle/replace the key in later upgrades or for planned signing key changes.)
+-->
+
+2.2. The application MUST support images singed by multiple keys.
+
+<!--
+* Public key TOFU (trust on first use): yes, if confirmed in the firmware.
+-->
+
+2.3. The application MUST ask the user for confirmation to trust given key
+     on the first use. The application MAY maintain the base of trusted keys
+     separately from UEFI Secure Boot or use existing UEFI Secure Boot
+     databases.
+
+<!--
+* Show key fingerprint: yes - "key fingerprint [key fingerprint]" [info symbol] "What is this? Brief explanation that the user can check with the vendor of this operating system if they wish to trust that key."
+-->
+
+2.4. The application SHOULD present the information about keys and
+     certificates in human-readable form using Issuer's and Subject's Common
+     Name (CN) and Organization (O) or other fields if present. Additionally,
+     the key fingerprint MUST always be displayed.
+
+<!--
+* Microsoft Secure Boot default signing key:
+** included in the firmware by default: yes, required for the purpose of allowing the user to boot Microsoft Windows
+** trusted in the firmware by default: no, express permission by the user is needed. (Because this key could lead to Secure Booting a Linux malware distribution. This would be unwanted by users who wish to use full verified boot with their chosen key.)
+* What this is:
+** A mechanism for users to much more easily choose which keys they trust.
+-->
+
+2.5. The firmware SHOULD include the Microsoft UEFI Secure Boot keys and
+     certificates in the firmware vendor's default set of UEFI Secure Boot
+     keys.
+
+2.6. The firmware vendor's default UEFI Secure Boot key and certificate set
+     SHALL NOT be trusted by default. Trust shall only be given to these keys
+     if explicitly confirmed by the user.
+
+<!--
+* What this is:
+** A mechanism for users to much more easily choose which keys they trust.
+-->
+
+2.7. The application SHOULD assist in trusted key/certificate set selection.
+
+<!--
+** If verification fails, load firmware-internal database of well-known keys that the user may or may not trust
+** Check boot medium for public key database in a well-known location; if it exists, load keys from it and add any new keys not already present in firmware to the list of possibly usable keys (but do not trust any key ever without user confirmation)
+-->
+
+2.8. The application SHALL scan EFI System Partition (ESP) for available keys
+     and certificates. The certificates MUST be X509 DER encoded. The
+     application MAY save the list of found and usable keys and certificates
+     in a separate UEFI variable for future use.
```

Comment:
Is there any reason why we limit to X.509? If we use LoadImage, we get automatic support for all keys. Unless we have some security considerations in mind, we should not limit ourselves.

We're talking about keys or certificates. What keys do we have in mind? If we support X.509, then we are talking only about certificates.

This requirement should be split and clarified. I think there are 3+ requirements in this one bullet.

Reply:
EDK2 supports only specific formats to be enrolled for PK or other variables. That's why.

Reply:
8f6adaf split